truffleruby: not supported? flaky specs? bad specs?

I noticed this failure in CI, it ran on 2302dbc658d013762be7d8c3a3759d224956d4a7 but it passed on the next commit (and I haven't seen it before).

https://github.com/sinatra/mustermann/actions/runs/10680826128/job/29603153585#step:7:10 (link will eventually not show any logs)

```
Failures:

  1) Mustermann::Expander hash is expected not to be == 2122681340268910596
     Failure/Error: example { Mustermann::Expander.new('/foo')               .hash.should_not be == Mustermann::Expander.new('/bar').hash }
       `expect(2122681340268910596).not_to be == 2122681340268910596`
     # ./mustermann/spec/expander_spec.rb:105:in `block (3 levels) in <top (required)>'

Finished in 5.15 seconds (files took 4.76 seconds to load)
4011 examples, 1 failure

Failed examples:

rspec ./mustermann/spec/expander_spec.rb:105 # Mustermann::Expander hash is expected not to be == 2122681340268910596

rake aborted!
Command failed with status (1): [/home/runner/.rubies/truffleruby-[24](https://github.com/sinatra/mustermann/actions/runs/10680826128/job/29603153585#step:7:25).0.2/bin/truffleruby -S rspec]
/home/runner/work/mustermann/mustermann/Rakefile:8:in `block in <top (required)>'
/home/runner/.rubies/truffleruby-24.0.2/bin/bundle:44:in `<main>'
Tasks: TOP => default => rspec
(See full trace by running task with --trace)
Error: Process completed with exit code 1.
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -31,8 +31,8 @@ jobs:
           - { ruby: head, sinatra: head }
           - { ruby: jruby, sinatra: ~> 4.0.0 }
           - { ruby: jruby-head, sinatra: head }
-          # https://github.com/sinatra/mustermann/issues/143
-          - { ruby: truffleruby, sinatra: ~> 4.0.0, allow-failure: true }
+          - { ruby: truffleruby, sinatra: ~> 4.0.0 }
+          - { ruby: truffleruby-head, sinatra: head }
     env:
       sinatra: ${{ matrix.sinatra }}
     steps:

